core/conversion/converters/impl/element_wise.cpp

```diff
@@ -9,246 +9,326 @@ namespace converters {
-      LOG_DEBUG(
-          "Input shape is less than 4D got: "
-          << util::toDims(shape) << ", inserting shuffle layers to reshape to 4D tensor shape: " << new_shape);
-      auto other_shuffle = ctx->net->addShuffle(*other);
-      other_shuffle->setReshapeDimensions(new_shape);
-      other_shuffle->setName(std::string("[Reshape other to " + util::toStr(new_shape) + ']').c_str());
-      other = other_shuffle->getOutput(0);
-
-      auto self_shuffle = ctx->net->addShuffle(*self);
-      self_shuffle->setReshapeDimensions(new_shape);
-      self_shuffle->setName(std::string("[Reshape self to " + util::toStr(new_shape) + ']').c_str());
-      self = self_shuffle->getOutput(0);
+nvinfer1::ILayer* add_elementwise(ConversionCtx* ctx,
+                                  nvinfer1::ElementWiseOperation op,
+                                  nvinfer1::ITensor* self,
+                                  nvinfer1::ITensor* other,
+                                  const std::string& name) {
+    if (self->getDimensions().nbDims < other->getDimensions().nbDims) {
+        std::swap(self, other);
     }
-
-    auto scale = Weights(ctx, scalar);
-    auto scaled = ctx->net->addScaleNd(*other, nvinfer1::ScaleMode::kUNIFORM, {}, scale.data, {}, 0);
-    auto scaled_other = scaled->getOutput(0);
-
-    // if (shape.size() < 4) {
-    //      LOG_DEBUG("Input shape wass less than 3D got, so was reshaped for
-    //      scale, reshaping back to: " << util::toDims(shape)); auto shuffle =
-    //      ctx->net->addShuffle(*scaled_other);
-    //      shuffle->setReshapeDimensions(util::toDims(shape));
-    //      shuffle->setName(std::string("[Reshape other to " +
-    //      util::toStr(util::toDims(shape)) + ']').c_str()); scaled_other =
-    //      shuffle->getOutput(0);
-    // }
-
-    ele = ctx->net->addElementWise(*self, *scaled_other, op);
-  } else {
-    ele = ctx->net->addElementWise(*self, *other, op);
-  }
-
-  return ele;
+    auto selfDim = util::toVec(self->getDimensions());
+    auto otherDim = util::toVec(other->getDimensions());
+    if (selfDim.size() != otherDim.size()) {
+        if (otherDim.end() != std::find(otherDim.begin(), otherDim.end(), -1)) {
+            auto otherShapeT = torch::ones(selfDim.size(), torch::kInt32);
+            auto selfShapeM = torch::zeros(selfDim.size(), torch::kInt32);
+            for (size_t start = selfDim.size() - otherDim.size(), idx = 0;
+                 idx < otherDim.size(); ++idx) {
+                if (-1 != otherDim[idx]) {
+                    otherShapeT[start + idx] = otherDim[idx];
+                } else {
+                    otherShapeT[start + idx] = 0;
+                    selfShapeM[start + idx] = 1;
+                }
+            }
+            auto otherShapeTX = tensor_to_const(ctx, otherShapeT);
+            auto selfShapeMX = tensor_to_const(ctx, selfShapeM);
+            auto selfShape = ctx->net->addShape(*self)->getOutput(0);
+            auto s1 =
```

Comment:
Variable names `s1` and `s2` can be renamed maybe to better reflect the function ?

Reply:
updated